src/hooks/useSmartSearch.ts

```diff
@@ -26,6 +34,17 @@ const useSmartSearch = (initialFilters: InitialFilters = []): UseSmartSearch =>
         ]);
     };
 
+    const addFilterToStart = (filter: ZetkinSmartSearchFilter) => {
```

Comment:
What is the purpose of this function? It looks like a general-purpose "prepend" function, but I don't think that's something that we need. I do realize it's necessary to be able to work with the `all` (or no) filter at the beginning of the list, but I would imagine a `setStartFilter()` (or `setStartWith()`) function to which the user of the hook could pass either the `all` filter or `null`, and then the logic in here could decide whether to add the filter, update an existing filter, remove it, or whatever.

In fact because the "starts with" state is special, it wouldn't need to be represented by a regular filter in the normal array at all, unless that's the easiest way to do it.

The idea is to move complexity out of the component and into this hook. Do you think that it would?